352 - IDS Error Page

jmacaluso711 opened the pull request:
**Explain the details for making this change. What existing problem does the pull request solve?**
This PR adds the IDS Error page component

**Related github/jira issue (required)**:
closes #352 

**Steps necessary to review your pull request (required)**:
- Pull and run branch
- test http://localhost:4300/ids-error-page/ 
- compare to [enterprise](https://main-enterprise.demo.design.infor.com/components/error-page/example-index.html)
- test "Return" button closes the error page
- make sure there is sufficient test coverage

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

<!-- After submitting your PR, please check back to make sure checks on the PR -->

vivz753 commented on src/components/ids-error-page/ids-error-page.js:
```diff
@@ -0,0 +1,234 @@
+  template() {
+    return `<ids-popup part="modal" class="ids-modal ids-error" type="custom" position-style="viewport">
+      <div class="ids-modal-container" slot="content">
+        <ids-empty-message icon="${!this.icon ? DEFAULT_ICON : this.icon}">
```
Consider using `this.icon ?? DEFAULT_ICON` instead of the ternary

vivz753 commented on src/components/ids-error-page/ids-error-page.js:
```diff
@@ -0,0 +1,234 @@
+            label="true"
+            slot="label"
+          >
+            ${!this.label ? 'Add Label' : this.label}
```
Same here. Consider using `this.label ?? 'Add Label'`

vivz753 commented on src/components/ids-error-page/ids-error-page.js:
```diff
@@ -0,0 +1,234 @@
+            ${!this.label ? 'Add Label' : this.label}
+          </ids-text>
+          <ids-text label="true" slot="description">
+            ${!this.description ? 'Add Description' : this.description}
```
And here

vivz753 commented on src/components/ids-error-page/ids-error-page.js:
```diff
@@ -0,0 +1,234 @@
+            ${!this.description ? 'Add Description' : this.description}
+          </ids-text>
+          <ids-button class="action-button" slot="button" type="primary">
+            <span slot="text">${!this.buttonText ? 'Action' : this.buttonText}</span>
```
You know what I'm going to say

vivz753 commented on src/components/ids-error-page/ids-error-page.js:
```diff
@@ -0,0 +1,234 @@
+      this.removeAttribute(attributes.ICON);
+    }
+
+    const emptyMessage = this.shadowRoot.querySelector('ids-empty-message');
```
Consider standardizing the use of `this.container`

vivz753 commented on src/components/ids-error-page/ids-error-page.js:
```diff
@@ -0,0 +1,234 @@
+   * @private
+   */
+  #attachEventHandlers() {
+    const button = this.shadowRoot.querySelector('.action-button');
```
Same here for `this.container`

vivz753 commented on src/components/ids-error-page/ids-error-page.js:
```diff
@@ -0,0 +1,234 @@
+      });
+    });
+
+    this.onEvent('touchstart', button, (e) => {
```
Is `touchstart` fired when the user first clicks, before releasing? I've had some issues where 'click' doesn't fire until after I lift my finger up off the mouse button

vivz753 commented on src/components/ids-error-page/ids-error-page.js:
```diff
@@ -0,0 +1,234 @@
+   */
+  get ariaLabelContent() {
+    const label = this.querySelector('[slot="label"')?.innerText;
+    return `${label || ''} ${this.label || ''} ${this.description || ''}`;
```
Consider using `??` instead of `||`

danielortiz1982 replied:
@vivz753 Could you possibly provide some more insight for @jmacaluso711 on this suggestion.

vivz753 reviewed (dismissed):
Looks good! My comments are mostly just preferences in syntax, but if you choose not to go with them, the component will function mostly the same.

tmcconechy commented on demos/ids-error-page/example.html:
```diff
@@ -0,0 +1,18 @@
+      visible="true"
+      icon="empty-error-loading"
+      label="Access Denied"
+      description="SunSystems Configuration Manager access has been denied."
```
```suggestion
      description="Configuration Manager access has been denied."
```

Just thinking about removing specific names

tmcconechy commented on src/components/ids-error-page/README.md:
```diff
@@ -0,0 +1,37 @@
+## Ids Error Page Component
+
+This example shows how to render a page in an IDS Error Page that shows an unknown error. Ideally its best to avoid these types of errors in your application but when an unexpected error occurs you can use this component to render information about the error.
```
```suggestion
This example shows how to render a page in an IDS Error Page that shows an unknown error. Ideally its best to avoid these types of errors in your application but when an unexpected error occurs you can use this component to render information about the error. The general use case is similar to a 40(x) error page.
```

tmcconechy commented on src/components/ids-error-page/ids-error-page.d.ts:
```diff
@@ -0,0 +1,15 @@
+  /* id of the icon to be displayed */
+  icon: string;
+
+  /* label of te error page to be displayed */
```
```suggestion
  /* label of the error page to be displayed */
```

tmcconechy commented on src/components/ids-error-page/ids-error-page.scss:
```diff
@@ -0,0 +1 @@
+// Error page styles
```
Maybe not needed?

tmcconechy commented on src/components/ids-error-page/index.js:
```diff
@@ -0,0 +1 @@
+export { default } from './ids-error-page';
```
The index file isnt needed anymore after https://github.com/infor-design/enterprise-wc/pull/462
But I'll sync this up once the first of these is merged

tmcconechy commented on test/ids-error-page/ids-error-page-func-test.js.snap:
```diff
@@ -0,0 +1,19 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Ids Error Page Tests renders correctly 1`] = `
+"<ids-overlay></ids-overlay><style nonce=\\"undefined\\">:host { background-color: transparent; }</style><ids-popup part=\\"modal\\" class=\\"ids-modal ids-error\\" type=\\"custom\\" position-style=\\"viewport\\" language=\\"en\\" locale=\\"en-US\\" align=\\"center\\">
```
Could set this to snap `ids-overlay.shadowRoot` (un-needed css and nonce)

tmcconechy reviewed (changes requested):
Just some minor things but overall looks good. In addition to the inline comments.

1. I noticed the old one doesnt animate on load maybe this one should also not animate
2. how do we change theme in the demo? I see in the percy image that the dark ICON is hard to see and I'm wondering if its the same as https://main-enterprise.demo.design.infor.com/components/error-page/example-index.html?theme=new&mode=dark so maybe we need to adjust that icon color in dark?

tmcconechy commented on demos/ids-error-page/example.html:
```diff
@@ -0,0 +1,18 @@
+<ids-layout-grid auto="true">
+  <ids-layout-grid-cell>
+    <ids-text font-size="12" type="h1">IDS Error Page</ids-text>
```
In this case maybe we dont need?

jmacaluso711 commented:
@tmcconechy
> how do we change theme in the demo? I see in the percy image that the dark ICON is hard to see and I'm wondering if its the same as https://main-enterprise.demo.design.infor.com/components/error-page/example-index.html?theme=new&mode=dark so maybe we need to adjust that icon color in dark?

I think this is handled in http://localhost:4300/ids-empty-message

@vivz753 I addressed your comments as well, thanks

tmcconechy commented:
@jmacaluso711 ok i see its the same on https://main-enterprise.demo.design.infor.com/components/emptymessage/example-widgets.html?theme=new&mode=dark&colors=0066D4 as http://localhost:4300/ids-empty-message

It looks better than i see in the percy image so i think we can leave that for now
